performance/collector/es.py

```diff
@@ -0,0 +1,89 @@
+
+import json
+import requests
+import six
+import os
+
+
+
+from six.moves.urllib import parse as urlparse
+from datetime import datetime
+
+
+from rally import api
+from rally.common import logging
+from rally import exceptions
+from rally.task import exporter
+
+LOG = logging.getLogger(__name__)
+
+
+@exporter.configure(name="es")
+class ESExporter(exporter.Exporter):
+
+    def validate(self):
+        """Validate connection string.
+
+        The format of connection string in ElasticSearch plugin is
+            es:///<path>.<region>
+        """
+        print(self.region)
+        print(self.url)
+
+        parse_obj = urlparse.urlparse(self.connection_string)
+        print(parse_obj)
+
+        if self.connection_string is None or parse_obj.path == "":
+            raise exceptions.InvalidConnectionString(
+                "It should be `es:///<ip_or_hostname>.<region>`.")
+        if not self.region:
+            raise exceptions.InvalidConnectionString(
+                "Invalid or undefined region")
+
+    def __init__(self, connection_string):
+        super(ESExporter, self).__init__(connection_string)
+        self.path = os.path.expanduser(urlparse.urlparse(
+            connection_string).path[1:])
+        self.region = connection_string.split(".")[-1]
+        self.url = connection_string.replace('.' + self.region,'').split('/')[-1]
+        self.proto = 'http://'
+        self.validate()
+
+    def export(self, uuid):
+        """Export results of the task to the file.
+
+        :param uuid: uuid of the task object
+        """
+
+        task = api.Task.get(uuid)
+
+        LOG.debug("Got the task object by it's uuid %s. " % uuid)
+
+        task_results = [{"key": x["key"], "result": x["data"]["raw"],
+                         "sla": x["data"]["sla"],
+                         "hooks": x["data"].get("hooks"),
+                         "load_duration": x["data"]["load_duration"],
+                         "full_duration": x["data"]["full_duration"]}
+                        for x in task.get_results()]
+
+        es_data_list = ""
```

Comment:
This doesn't look good for many reasons: 

- First of all in python you shouldn't say in variable what type it is 
- You are saying that it's list and it's sting!
- String operations in python are very slow, you'll recreate string each time, way better idea is to create array and join it to string once, take a look here for sample: https://github.com/seecloud/health/blob/master/extra/es_gen_ts_index.py#L34-L45